addons/website_livechat/static/src/models/visitor/visitor.js

```diff
@@ -0,0 +1,133 @@
+odoo.define('website_livechat/static/src/models/partner/partner.js', function (require) {
+'use strict';
+
+const { registerNewModel } = require('mail/static/src/model/model_core.js');
+const { attr, many2one, one2many } = require('mail/static/src/model/model_field.js');
+
+function factory(dependencies) {
+
+    class Visitor extends dependencies['mail.model'] {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('country_id' in data && data.country_id) {
+                data2.country = [['insert', {
+                    id: data.country_id,
+                    code: data.country_code,
+                }]];
+            }
+            if ('history' in data) {
+                data2.history = data.history;
+            }
+            if ('is_connected' in data) {
+                data2.is_connected = data.is_connected;
+            }
+            if ('lang' in data) {
+                data2.lang = data.lang;
+            }
+            if ('name' in data) {
+                data2.name = data.name;
+            }
+            if ('partner_id' in data) {
+                if (data.partner_id) {
+                    data2.partner = [['insert', { id: data.partner_id }]];
+                } else {
+                    data2.partner = [['unlink-all']];
+                }
+            }
+            if ('website' in data) {
+                data2.website = data.website;
+            }
+            return data2;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAvatarSrc() {
+            if (!this.partner) {
+                return '/mail/static/src/img/smiley/avatar.jpg';
+            }
+            return `/web/image/res.partner/${this.partner.id}/image_128`;
+        }
+
+        /**
+         * @private
+         * @returns {mail.country}
+         */
+        _computeCountry() {
+            if (this.partner && this.partner.country) {
+                return [['link', this.partner.country]];
+            }
+            if (this.country) {
+                return [['link', this.country]];
+            }
+            return [['unlink-all']];
```

Comment:
Just `unlink` for x2one

Reply:
without argument ? `[['unlink']]` ?

Reply:
Correct, we have plenty of them. There is only one `unlink` possibility for x2one, other arguments are ignored (see `_setRelationUnlinkX2One` has no value param).